src/components/Wizard/CreateVmWizard/providers/vmwareActions.js

```diff
@@ -126,6 +128,14 @@ export const onVCenterInstanceSelected = async (
   );
 };
 
+const prefillVmName = (basicSettings, onFormChange, vmName, formValid) => {
+  const formName = settingsValue(basicSettings, NAME_KEY);
+  const prevSelectedVmName = settingsValue(basicSettings, PROVIDER_VMWARE_VM_KEY);
+  if (!formName || formName === prevSelectedVmName) {
+    onFormChange({ value: vmName }, NAME_KEY, formValid);
+  }
+};
+
 export const onVCenterVmSelectedConnected = async (
```

Comment:
just a suggestion: wouldn't it be better to move these callbacks from each field onChange to `publish()` in BasicSettingsTab and resolve everything there according to KEY?

Reply:
I would like to avoid that as it is splitting business logic across multiple component and especially layers. Good for now, but harder to maintain later. I prefer collocating related things.